### DISCUSSION TITLE
Enabling and Disabling the Task Before the DAG Run From UI and API

### DISCUSSION THREAD

**Discussion by Amar1404** [apache/airflow#26825] [General]
### Description

Enabling and Disabling the task for the particular DAG from the UI and API. or using the parameter that disable it. This was it is easy to disable a particular task easily while running manually or schedule. The UI gives us the feature to select the task to disable it before triggering it manually

### Use case/motivation

So  for the backfill or running manually in a DAG if we don't want to run a particular task we can disable it, from UI and in the next schedule the particular task will not run. That way we don't have to remove the code for the task.

### Are you willing to submit a PR?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by potiuk** [comment]
I think this one will be hard to implement and it is not something we really want to do. Tasks have dependencies to other tasks, and it might get tricky to decide what to do when one or more tasks are disabled. 

But I wonder what others think,

**Comment by potiuk** [comment]
Not much interest, I see, convertting that into a discussion, maybe someone will pick an interest to discuss about it and turn it into actual feature/PR?

**Comment by manish-persistent** [comment]
This is a required feature. Many a times we need to run only a single task and once confirmed it has done its intended work, then manually run/disable downstream task.

**Reply by potiuk** [reply]
You can use `airflow tasks run` command for that - UI is not really to "do" stuff but to manage history f runs. CLI is better for any kind of operations that require manual messing with tasks/dags (backfill etc). 

 https://airflow.apache.org/docs/apache-airflow/stable/cli-and-env-variables-ref.html#run

**Reply by Amar1404** [reply]
In some companies like Devops don't provide command like access and in EKS like setup. It's very tedious to run a task with command line, where multiple teams are involved.And Second lets we want to trigger task with particular DAG config, and there are multiple case.

And It will bring many Azkaban users to migrate to Airflow.

**Reply by potiuk** [reply]
Then you need to have a test/staging environment that you should use for testing. Running a single task to verify if it runs on a production system is a very bad idea. You need to setup a dev/staging system for that. 

If you want you can also develop a UI plugin to run "airflow tasks test" under the hood. This should be rather easy to do.

I doubt we will ever add a "tesstin" workflow to our UI - this has never been intended use case for the UI.

But if you think it is really needed for your case - feel free to develop plugin to do it (you can even publish it for others).
